book-store/book-store.js

```diff
@@ -0,0 +1,132 @@
+            }
+        },
+    };
+
+    let book_2 = {
+        name: "HTML. The Definitive Guide",
+        price: 12,
+        pageCount: 128,
+        image: "img/html.jpg",
+        rating: {
+            reader: 3,
+            critic: 5,
+            getAverageRating() {
+                return Math.ceil((this.reader + this.critic) / 2)
+            }
+        },
+    };
+
+    let book_3 = {
+        name: "JavaScript: The Definitive Guide",
+        price: 50,
+        pageCount: 280,
+        image: "img/java_script.jpg",
+        rating: {
+            reader: 4,
+            critic: 4,
+            getAverageRating() {
+                return Math.ceil((this.reader + this.critic) / 2)
+            }
+        }
+    };
+
+    let book_4 = {
+        name: "Zombie Raccoons & Killer Bunnies",
+        price: 20,
+        pageCount: 460,
+        image: "img/zombie_raccoons.jpg",
+        rating: {
+            reader: 2,
+            critic: 1,
+            getAverageRating() {
+                return Math.ceil((this.reader + this.critic) / 2)
+            }
+        },
+    };
+
+    let bookList = document.querySelector('.book-list');
+    let books = [book_1, book_2, book_3, book_4];
+
+    let getBookDescription = (rating) => {
+        let stars = "";
+        for (let i = 0; i < rating; i++) {
+            stars = stars + "⭐";
+        }
+        return `${stars}
+         This book got an average rating of ${rating} stars based on readers and critics evaluations
+         
+         `
+    };
+
+    function generateBookElWithHTML(book) {
+        let bookEl = document.createElement('div');
+        bookEl.classList.add("book");
+        bookEl.innerHTML = `
+                         <div class= 'book__img'>
+                            <img  src="${book.image}">
+                        </div>
+                        <div class="book__info">
+                            <div class="book__name"> ${book.name}</div>
+                            <div class="book__pages">${book.pageCount} pages</div>
+                            <div class="book__review">${getBookDescription(book.rating.getAverageRating())}</div>
+                            <button class="book__price">Buy it now for ${book.price} €</button>
+                        </div>
+                `;
+        return bookEl;
+    }
+
+    function generateBookElWithJS(book) {
+        let bookEl = document.createElement('div');
+
+        let bookImageWrapper = document.createElement('div');
+        let bookImage = document.createElement('img');
+        bookImage.classList.add('book__img');
+        bookImageWrapper.appendChild(bookImage);
+        bookImage.src = book.image;
+        bookEl.appendChild(bookImageWrapper);
+
+        let bookInfo = document.createElement('div');
+        bookInfo.classList.add('book__info');
+        bookEl.appendChild(bookInfo);
+
+        let bookName = document.createElement('div');
```

Comment:
we can very far and create a separate function to generate name, pageCount and all other elements for main object.